nzshm_hazlab/data/data_loaders/thp_loader.py

```diff
@@ -0,0 +1,167 @@
+"""This module provides the THPHazardLoader class for producing hazard curves from dynamically created hazard models."""
+
+import functools
+from typing import TYPE_CHECKING
+
+import numpy as np
+import toshi_hazard_post.aggregation_calc as aggregation
+import toshi_hazard_post.calculators as calculators
+import toshi_hazard_post.data as data
+from toshi_hazard_post.logic_tree import HazardLogicTree
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    import pyarrow as pa
+    import pyarrow.dataset as ds
+    from nzshm_common import CodedLocation
+    from nzshm_model.logic_tree import GMCMLogicTree, SourceLogicTree
+
+
+@functools.cache
+def _get_batch_table_cached(
+    dataset: 'ds.Dataset',
+    compatible_calc_id: str,
+    sources_digests: tuple[str],
+    gmms_digests: tuple[str],
+    nloc_0: str,
+    vs30: int,
+    imts: tuple[str],
+) -> 'pa.Table':
+
+    return data.get_batch_table(dataset, compatible_calc_id, sources_digests, gmms_digests, vs30, nloc_0, imts)
+
+
+class THPHazardLoader:
+    """A class for creating hazard curves from user-defined hazard models.
+
+    This class allows a user to calculated hazard curves from  a hazard model constructed by defining source
```

Comment:
The first sentence doesn't seem to parse

Reply:
fixed.